core/controllers/admin.py

```diff
@@ -245,6 +250,19 @@ def post(self):
                 result = {
                     'opportunities_count': opportunities_count
                 }
+            elif self.payload.get('action') == 'update_feature_flag_rules':
+                feature_name = self.payload.get('feature_name')
+                new_rule_dicts = self.payload.get('new_rules')
+                commit_message = self.payload.get('commit_message')
+                try:
+                    platform_feature_services.update_feature_flag_rules(
+                        feature_name, self.user_id, commit_message,
+                        new_rule_dicts)
+                except Exception as e:
```

Comment:
Should we validate all the payloads types before passing it to `update_feature_flag_rules`? 

Currently, it will through an invalid input exception for any error, right? This sometimes makes it harder to understand the reason for failure! In past, we have tried to either to have a small scope of exception (like except ValidationErrorException) or remove it completely!

Reply:
Done, added type checks for the payload members. Also modified the logics for the exceptions, now it catches all `utils.ValidationError` caused by invalid rules and `Exception` caused by unknown names, all other Exceptions will be raised again.